### ISSUE TITLE
possible update gcis:hasVersion

### ISSUE THREAD

**Issue by zednis** [USGCRP/gcis-ontology#18]
The domain of `gcis:hasVersion` is `gcis:Model`.

```
gcis:hasVersion a owl:DatatypeProperty ;
    rdfs:label "Has Version" ;
    rdfs:comment "A model may have a version." ;
    rdfs:domain gcis:Model ;
    rdfs:range xsd:string ;
    rdfs:subPropertyOf dcterms:hasVersion .
```

From the property's comment it is clear this property is intended specifically for the model version, but it has a very broad name and label.

I think we should do one of the following:
1) updating the property to be `gcis:modelVersion`
2) broadening the definition of `gcis:hasVersion` to not be just for model versions
3) removing the property entirely and using `dcterms:hasVersion` directly in the instance data.

**Comment by rewolfe** [comment]
I agree that the version domain should be for more than just the model.  It
is also applicable to datasets and activities (e.g. software version).

On Tue, Jun 16, 2015 at 5:33 PM, Stephan Zednik notifications@github.com
wrote:

> The domain of gcis:hasVersion is gcis:Model.
> 
> gcis:hasVersion a owl:DatatypeProperty ;
>     rdfs:label "Has Version" ;
>     rdfs:comment "A model may have a version." ;
>     rdfs:domain gcis:Model ;
>     rdfs:range xsd:string ;
>     rdfs:subPropertyOf dcterms:hasVersion .
> 
> From the property's comment it is clear this property is intended
> specifically for the model version, but it has a very broad name and label.
> 
> I think we should do one of the following:
> 1) updating the property to be gcis:modelVersion
> 2) broadening the definition of gcis:hasVersion to not be just for model
> versions
> 3) removing the property entirely and using dcterms:hasVersion in the
> instance data.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/18.

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by bduggan** [comment]
> I think we should do one of the following:
> 1) updating the property to be `gcis:modelVersion`
> 2) broadening the definition of `gcis:hasVersion` to not be just for model versions
> 3) removing the property entirely and using dcterms:hasVersion in the instance data.

My impression is that model versions are very distinct from one another,
so much so that the versions are often built into the identifiers.
Different model versions may have different attributes. Some examples
are at http://data.globalchange.gov/model and also
the draft health report [appendix](http://www.globalchange.gov/health-assessment),
table A1.5.

For these reasons "modelVersion" may be more appropriate.

**Comment by zednis** [comment]
How about option 3, removing the `gcis:hasVersion` from the ontology and using `dcterms:hasVersion` in the RDF?

edit: `dcterms:hasVersion` does not currently have a range and is not specified as an object property or as a datatype property.  How about we keep `gcis:hasVersion` as we have specified it as a datatype property with range of string and instead remove the domain entirely so anything can be given a version?

So, how about option 2 ?

**Comment by bduggan** [comment]
On Wednesday, June 17, Stephan Zednik wrote: 

> How about option 3, removing the `gcis:hasVersion` from the ontology and using dcterms:hasVersion in the RDF?

"A related resource that is a version, edition, or adaptation of the described resource."

Would this mean that we would be undertaking to trace which models are versions of which other models?  That would be interesting but could be hard (e.g. is [CCSM4](http://www.cesm.ucar.edu/models/ccsm4.0/) a version of CESM1?) Another notion of version would just be a particular label/tag applied to a set of algorithms.

**Comment by zednis** [comment]
@bduggan good catch.  On further look it appears that the `dcterms:hasVersion` is similar to the frbr and fabio relationships  that connect (work -> expression -> manifestation -> item) and is not a simple datatype to express version information.

With this in mind I would not recommend using `dcterms:hasVersion`.

We could use `<http://dbpedia.org/property/version>` (see http://dbpedia.org/data4/version.n3)

**Comment by bduggan** [comment]
On Wednesday, June 17, Stephan Zednik wrote: 

> We could use `<http://dbpedia.org/property/version>` (see http://dbpedia.org/data4/version.n3)

That looks good.

**Comment by justgo129** [comment]
fine here

On Wed, Jun 17, 2015 at 3:13 PM, Brian Duggan notifications@github.com
wrote:

> On Wednesday, June 17, Stephan Zednik wrote:
> 
> > We could use `<http://dbpedia.org/property/version>` (see
> > http://dbpedia.org/data4/version.n3)
> 
> That looks good.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/18#issuecomment-112917299
> .

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by rewolfe** [comment]
Looks good.

On Wed, Jun 17, 2015 at 3:18 PM, justgo129 notifications@github.com wrote:

> fine here
> 
> On Wed, Jun 17, 2015 at 3:13 PM, Brian Duggan notifications@github.com
> wrote:
> 
> > On Wednesday, June 17, Stephan Zednik wrote:
> > 
> > > We could use `<http://dbpedia.org/property/version>` (see
> > > http://dbpedia.org/data4/version.n3)
> > 
> > That looks good.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/USGCRP/gcis-ontology/issues/18#issuecomment-112917299>
> > .
> 
> ## 
> 
> ---
> 
> Justin Goldstein, Ph.D.
> Advance Science Climate Data and Observing Systems Coordinator
> US Global Change Research Program
> 1717 Pennsylvania Ave NW, Suite #250
> Washington, DC 20006
> 
> O: (202) 419-3496
> M: (202) 285-3005
> 
> e-mail: jgoldstein AT usgcrp Dot gov
> http://www.globalchange.gov
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/18#issuecomment-112918764
> .

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by zednis** [comment]
I propose the following changes:
1) remove `gcis:hasVersion` from the GCIS ontology
2) update the RDF templates to use `<http://dbpedia.org/property/version>` (I need to check on the prefix)

edit - I think there is already general agreement on this, I just wanted to make the proposed change official.

**Comment by bduggan** [comment]
:+1:

**Comment by justgo129** [comment]
I'm fine with it, but could someone confirm that the domains/ranges of <
http://dbpedia.org/property/version> match our instance data?  I don't see
one listed there.  Explicitly, are both floats and integers encompassed?

On Tue, Jun 23, 2015 at 4:00 PM, Brian Duggan notifications@github.com
wrote:

> :+1
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/18#issuecomment-114624305
> .

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by zednis** [comment]
@justgo129 This is what I got when I asked for the RDF of the property:

```
@prefix rdf:    <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix dbpprop:    <http://dbpedia.org/property/> .
dbpprop:version rdf:type    rdf:Property .
@prefix owl:    <http://www.w3.org/2002/07/owl#> .
dbpprop:version owl:sameAs  dbpprop:version .
@prefix rdfs:   <http://www.w3.org/2000/01/rdf-schema#> .
dbpprop:version rdfs:label  "version"@en .
```

It does not have a domain or range and is defined as a rdf:Property so it can be treated as an owl:ObjectProperty or owl:DatatypeProperty.  it can have any rdfs:Literal as a value.  The downside of this that it does not conform to OWL-DL.

If we want to use a property that confirms to OWL-DL then we should define the following:

```
gcis:version a owl:DatatypeProperty ;
  rdfs:label "version"@en ;
  rdfs:subPropertyOf dbpprop:version ;
```

That is OWL-DL and can have any rdfs:Literal value (string, integer, etc.)

**Comment by justgo129** [comment]
Excellent.  I'm fine with this then if @bduggan and @aulenbac are.  Your
explanation answered my question, Stephan -- thanks.

**Comment by xgmachina** [comment]
+1

**Comment by zednis** [comment]
resolved by #23
